pkg/controller/v1beta1/inferenceservice/reconcilers/kubernetes/kube_reconciler.go

```diff
@@ -0,0 +1,387 @@
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec,
+	componentStatus v1beta1.ComponentStatusSpec) *exv1beta1.Ingress {
+	ingressHost := generateIngressHost(client, componentMeta)
+	pathType := exv1beta1.PathType(exv1beta1.PathTypeImplementationSpecific)
+	ingress := &exv1beta1.Ingress{
+		TypeMeta: metav1.TypeMeta{
+			Kind: string(exv1beta1.PolicyTypeIngress),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.GetNativeServiceIngressName(componentMeta.Name, componentMeta.Namespace),
+			Namespace: componentMeta.Namespace,
+		},
+		Spec: exv1beta1.IngressSpec{
+			Rules: []exv1beta1.IngressRule{
+				{
+					Host: ingressHost,
+					IngressRuleValue: exv1beta1.IngressRuleValue{
+						HTTP: &exv1beta1.HTTPIngressRuleValue{
+							Paths: []exv1beta1.HTTPIngressPath{
+								{
+									Path:     "/" + string(constants.Predict),
+									PathType: &pathType,
+									Backend: exv1beta1.IngressBackend{
+										ServiceName: componentMeta.Name,
+										ServicePort: intstr.IntOrString{
+											Type:   intstr.Int,
+											IntVal: podSpec.Containers[0].Ports[0].ContainerPort,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	return ingress
+}
+
+func getClientInCluster() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	return kubernetes.NewForConfig(config)
+}
+
+//Reconcile ...
+func (r *NativeReconciler) Reconcile() (*appsv1.Deployment, error) {
+	client, err := getClientInCluster()
+	if err != nil {
+		return nil, err
+	}
+
+	//reconcile Deployment
+	deployment := &appsv1.Deployment{}
+	checkResult, err := r.checkNativeDeploymentExist(client)
+	switch {
+	case checkResult == CheckResultCreate:
+		deployment, err = client.AppsV1().Deployments(r.Deployment.Namespace).Create(context.TODO(), r.Deployment, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultExisted:
+		deployment = r.Deployment
+	case checkResult == CheckResultUpdate:
+		deployment, err = client.AppsV1().Deployments(r.Deployment.Namespace).Update(context.TODO(), r.Deployment, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultUnknown:
+		return nil, err
+	}
+
+	//reconcile Service
+	checkResult, err = r.checkNativeServiceExist(client)
+	switch {
+	case checkResult == CheckResultCreate:
+		_, err = client.CoreV1().Services(r.Service.Namespace).Create(context.TODO(), r.Service, metav1.CreateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultExisted:
+		log.Info("Reconcile", "Service", "Existed. Will do nothing.")
+	case checkResult == CheckResultUpdate:
+		_, err = client.CoreV1().Services(r.Service.Namespace).Update(context.TODO(), r.Service, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, err
+		}
+	case checkResult == CheckResultUnknown:
+		return nil, err
+	}
```

Comment:
I think we can simplify this block and avoid switch cases, we only need create and update, if there is an error just return back.